projects/spectator/src/lib/spectator-with-routing/activated-route-stub.ts

```diff
@@ -0,0 +1,95 @@
+import { convertToParamMap, ActivatedRoute, ActivatedRouteSnapshot, Data, Params, ParamMap } from '@angular/router';
+import { Observable, ReplaySubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { RouteOptions } from './route-options';
+
+/**
+ * @publicApi
+ *
+ * Utility class for stubbing ActivatedRoute of @angular/router
+ */
+export class ActivatedRouteStub extends ActivatedRoute {
+  private testParams: Params = {};
+  private testQueryParams: Params = {};
+  private testData: Data = {};
+
+  private paramsSubject = new ReplaySubject(1);
+  private queryParamsSubject = new ReplaySubject(1);
+  private dataSubject = new ReplaySubject(1);
+
+  constructor(options?: RouteOptions) {
+    super();
+
+    if (options) {
+      this.testParams = options.params || {};
+      this.testQueryParams = options.queryParams || {};
+      this.testData = options.data || {};
+    }
+
+    this.params = this.paramsSubject.asObservable();
```

Comment:
Do we need distinctUntilChanged()?

Reply:
No, I don't think so, because in the "real world" all observables get notified on a new navigation.